apm-agent-plugins/apm-apache-httpclient-plugin/src/test/java/co/elastic/apm/agent/httpclient/helper/RequestHeaderAccessorTest.java

```diff
@@ -0,0 +1,54 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2021 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.httpclient.helper;
+
+import co.elastic.apm.agent.impl.transaction.AbstractTextHeaderGetterTest;
+import org.apache.http.HttpRequest;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.message.BasicHttpRequest;
+import org.junit.jupiter.api.Nested;
+
+import java.util.List;
+import java.util.Map;
+
+class RequestHeaderAccessorTest {
+
+    @Nested
+    class GetterTest extends AbstractTextHeaderGetterTest<RequestHeaderAccessor, HttpRequest> {
```

Comment:
Why is this required?

Reply:
It's not really needed, but as we only test for the `Getter` part it made more sense to me this way:
- the abstract test works for all `TextHeaderGetter` implemenations, including those that are not `TextHeaderSetter`
- we can only inherit a single class, thus if we were to add an `AbstractTextHeaderSetterTest` that would be required to test for the `Setter` part.

But I agree that's not required with the current state. Do you think it's worth changing that or can we merge as-is ?

Reply:
Either way can work, but generally speaking - I prefer not having non-used code only for potential future usage. Maybe the person who adds these setter tests in the future will add them to the same abstract class you created and call it `AbstractTextHeaderAccessorTest`, in which case it will remain irrelevant.

> Do you think it's worth changing that or can we merge as-is ?

Since it's super-easy to change, I'd vote for that, but the PR is approved either way, so no need to.